url8d3c4d17f5_catseye_JaC64_tgz-pJ8-JaC64J8/src/com/dreamfabric/c64utils/Debugger.java

```diff
@@ -0,0 +1,2110 @@
+/**
+ * This file is a part of JaC64 - a Java C64 Emulator
+ * Main Developer: Joakim Eriksson (Dreamfabric.com)
+ * Contact: joakime@sics.se
+ * Web: http://www.dreamfabric.com/c64
+ * ---------------------------------------------------
+ */
+package com.dreamfabric.c64utils;
+import com.dreamfabric.jac64.*;
+
+/**
+ * @author jblok
+ * @author Joakim Eriksson (joakime@sics.se
+ */
+public class Debugger implements IMonitor,M6510Ops
+{
+  private boolean DEBUG = false;
+  private boolean DEBUG_IRQ = true;
+  private boolean noinstructions = false;
+  private int level = 10;
+  String prefix = "";
+  public long lastCycles = 0;
+
+
+  private MOS6510Core cpu;
+  private int[] memory;
+
+  public void init(MOS6510Core cpu) {
+    this.cpu = cpu;
+    memory = cpu.getMemory();
+  }
+
+  /**
+   * @see IMonitor#isEnabled()
+   */
+  public boolean isEnabled()
+  {
+    return DEBUG;
+  }
+
+  public void setPrefix(String prefix) {
+    this.prefix = prefix;
+  }
+
+  public void info(Object o) {
+    output((String) o);
+  }
+
+  public void warning(Object o) {
+    output((String) o);
+  }
+
+  public void error(Object o) {
+    output((String) o);
+  }
+
+  private void output(String s) {
+    if (prefix != null) {
+      if (s.startsWith(prefix)) {
+	System.out.println(s);
+      }
+    } else {
+      System.out.println(s);
+    }
+  }
+
+  private int fetchByte(int pc) {
+    return memory[pc];
+  }
+
+  /**
+   * Only for debugging purposes
+   * @see IMonitor#disAssemble(int, int, int, int, int, byte, boolean)
+   */
+  public void disAssemble(int[] memory,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 135 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.